### ISSUE TITLE
Step 9.3.1: "count(distinct hhid.HoHID + hhid.ActiveHHType)" possible issue in Sample Code

### ISSUE THREAD

**Issue by nataliesus** [HMIS/LSASampleCode#733]
Hi @MollyMcEvilley ,

I am not sure if "count(distinct hhid.HoHID + hhid.ActiveHHType)" can be used for distinct combinations of [HoHID] and [ActiveHHType].

Could you please double-check how it works for MSSql? Because in MariaDb the result will be a sum of [HoHID] and [ActiveHHType]. For next 4 different households:
- [HoHID] = 180 and  [ActiveHHType] = 1
- [HoHID] = 179 and  [ActiveHHType] = 2
- [HoHID] = 178 and  [ActiveHHType] = 3
- [HoHID] = 82 and  [ActiveHHType] = 99

`count(distinct hhid.HoHID + hhid.ActiveHHType)` = 1 (distinct of 181)

Best regards,
Natalie

**Comment by MollyMcEvilley** [comment]
Bah.  You are correct -- all of the HoHIDs in my db are type varchar but effectively integers, and MSSQL is ignoring the data type, casting HoHID as int,  and just adding HoHID and ActiveHHType rather than concatenating.

Modifying the sample code to specifically cast ActiveHHType to varchar so it will concatenate to a string.

  ```
select distinct case when rv.RowID = 53 then count(distinct n.PersonalID) 
			  else count(distinct hhid.HoHID + cast(hhid.ActiveHHType as varchar)) end
		  , rv.Cohort, rv.Universe, hhid.ActiveHHType, rp.PopID, rv.SystemPath
```

Commit forthcoming!

**Comment by MollyMcEvilley** [comment]
Resolved in commit  f54b07b
